ironfish/src/wallet/wallet.ts

```diff
@@ -1002,14 +1001,59 @@ export class Wallet {
     }
   }
 
-  private buildAmountsNeeded(
+  async fundFromNoteHashes(
     raw: RawTransaction,
     options: {
+      account: Account
+      noteHashes: Buffer[]
+      fee?: bigint
+      feeRate?: bigint
+    },
+  ): Promise<void> {
+    const amountsNeeded = this.buildAmountsNeeded(raw)
+    const amountsSpent = new AssetBalances()
+
+    for (const noteHash of options.noteHashes) {
+      const decryptedNote = await options.account.getDecryptedNote(noteHash)
+      Assert.isNotUndefined(decryptedNote)
+      Assert.isNotNull(decryptedNote.index)
+
+      const witness = await this.chain.notes.witness(decryptedNote.index)
```

Comment:
Is there a way we can make this share the same call that a normal fund transaction request makes? So we aren't adding another instance of calling the chain db to the wallet

Reply:
I think with some refactoring of the normal `fund` both will be this one-liner